smarts/core/tests/test_mutiple_traffic_data.py

```diff
@@ -69,6 +70,24 @@
 }
 
 
+@pytest.fixture
+def create_history_file_1(tmp_path):
+    d = tmp_path / "sub"
+    d.mkdir()
+    p = d / "traffic_history_1.json"
+    p.write_text(json.dumps(traffic_history_1))
+    return p
+
+
+@pytest.fixture
+def create_history_file_2(tmp_path):
+    d = tmp_path / "sub2"
+    d.mkdir()
+    p = d / "traffic_history_2.json"
+    p.write_text(json.dumps(traffic_history_2))
+    return p
```

Comment:
I think this could work as configuration, it would split the test into two but would reduce the complexity.

```suggestion
@pytest.fixture(params=[(traffic_history_1, "traffic_history_1"), (traffic_history_2, "traffic_history_2")])
def create_history_file(tmp_path, request):
    traffic_history, name = request.param
    d = tmp_path
    d.mkdir()
    path = d / f"{name}.json"
    path.write_text(json.dumps(traffic_history))
    return (traffic_history, path)
```

Then you can turn the test into this:

```
def test_mutiple_traffic_data(create_scenario, create_history_file):
    traffic_history, path = create_history_file
```

Reply:
I think this will cause `test_mutiple_traffic_data` to run twice each time with different traffic history file, right? The behavior we want from `Scenario.discover_traffic_histories` is so that it returns two history files at once in order to test if multiple traffic data will create multiple scenarios. 

I agree that we can simplify the two similar `create_history_file` though and I will work on that

Reply:
I think this will cause `test_mutiple_traffic_data` to run twice each time with different traffic history file, right? The behavior we want from `Scenario.discover_traffic_histories` is so that it returns two history files at once in order to test if multiple traffic data will create multiple scenarios. 

I agree that we can simplify the two similar `create_history_file` though and I will work on that